course2_react/todo-list/src/List.js

```diff
@@ -0,0 +1,38 @@
+const List = ({ names, deleteName, setNames, input, setInput }) => {
+  //const handeledIt = deleteName();
+
+  const addName = () => {
+    let newId = names.length === 0 ? 1 : names[names.length - 1].id + 1;
+
+    const tempObj = { id: newId, n1: input };
+    console.log(tempObj);
+    setNames([...names, tempObj]);
+    console.log(names);
+  };
+
+  const addOnChange = (event) => {
+    setInput(event.target.value);
+    console.log(input);
+  };
+
+  return (
+    <div className="list">
+      <input value={input} type="text" onChange={addOnChange} />
```

Comment:
One more thing. Just remember in react all components should be for a specific use & that component should receive only those props which are required.

In you case you have kept deleteName function in container & add name function here in List component? why so?

List component should only deal with showing the todo list. So it will recieve the names list & a delete name function as prop

The Input & add button can be kept in separate component called as AddTodo component or can be used directly in container where you have used List component

And finally functions like addName, deleteName etc should stay in container itself